packages/web/src/shared/hooks/useAuth.ts

```diff
@@ -0,0 +1,67 @@
+import { useQuery } from '@tanstack/react-query';
+import { GetSessionResponse, SystemRole } from '@translation/api-types';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import apiClient from '../apiClient';
+
+export interface UseAuthOptions {
+  authenticated?: boolean;
+  systemRoles?: SystemRole[];
+}
+
+export type UseAuthResult =
+  | {
+      status: 'authenticated';
+      user: Required<GetSessionResponse>['user'];
+    }
+  | { status: 'unauthenticated' | 'loading'; user?: undefined };
+
+export default function useAuth({
+  authenticated,
+  systemRoles,
+}: UseAuthOptions = {}): UseAuthResult {
+  const navigate = useNavigate();
+  const { data, status } = useQuery(['session'], async () =>
+    apiClient.getSession()
+  );
+
+  useEffect(() => {
+    if (status === 'success') {
+      const { user } = data;
+      if (user) {
+        if (
+          authenticated === false ||
```

Comment:
Why would an authenticated user not be allowed to see a certain page that authenticated users could see?

Reply:
I clarified in the comments, but the basic use case is for login/signup pages, or for unauthenticated versions of pages on different routes.